samples/bluetooth/mesh/sensor_client/src/model_handler.c

```diff
@@ -36,56 +33,112 @@ static void sensor_cli_data_cb(struct bt_mesh_sensor_cli *cli,
 	}
 }
 
-static void sensor_cli_series_entry_cb(
-	struct bt_mesh_sensor_cli *cli, struct bt_mesh_msg_ctx *ctx,
-	const struct bt_mesh_sensor_type *sensor, uint8_t index, uint8_t count,
-	const struct bt_mesh_sensor_series_entry *entry)
+static void sensor_cli_series_entry_cb(struct bt_mesh_sensor_cli *cli, struct bt_mesh_msg_ctx *ctx,
+				       const struct bt_mesh_sensor_type *sensor, uint8_t index,
+				       uint8_t count,
+				       const struct bt_mesh_sensor_series_entry *entry)
 {
-	printk("Relative runtime in %d to %d degrees: %s percent\n",
-	       entry->value[1].val1, entry->value[2].val1,
-	       bt_mesh_sensor_ch_str(&entry->value[0]));
+	printk("Relative runtime in %d to %d degrees: %s percent\n", entry->value[1].val1,
+	       entry->value[2].val1, bt_mesh_sensor_ch_str(&entry->value[0]));
+}
+
+static void sensor_cli_setting_status_cb(struct bt_mesh_sensor_cli *cli,
+					 struct bt_mesh_msg_ctx *ctx,
+					 const struct bt_mesh_sensor_type *sensor,
+					 const struct bt_mesh_sensor_setting_status *setting)
+{
+	printk("Sensor ID: 0x%04x, ", sensor->id);
+	printk("Setting ID: 0x%04x\n", setting->type->id);
+	for (int chan = 0; chan < setting->type->channel_count; chan++) {
+		printk("Channel %d value: %s\n", chan,
+		       bt_mesh_sensor_ch_str(&(setting->value[chan])));
+	}
 }
 
 static const struct bt_mesh_sensor_cli_handlers bt_mesh_sensor_cli_handlers = {
 	.data = sensor_cli_data_cb,
 	.series_entry = sensor_cli_series_entry_cb,
+	.setting_status = sensor_cli_setting_status_cb,
 };
 
-static struct bt_mesh_sensor_cli sensor_cli =
-	BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
+static struct bt_mesh_sensor_cli sensor_cli = BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
 
 static struct k_work_delayable get_data_work;
 
 static void get_data(struct k_work *work)
 {
```

Comment:
It would be nice if you could prevent attempts to send messages as long as the device is not provisioned. See `bt_mesh_is_provisioned()`. An example of how it can be used is present in the light_switch sample.